### DISCUSSION TITLE
Error with DistanceSensor "pull_up:False"

### DISCUSSION THREAD

**Discussion by carlestecno** [gpiozero/gpiozero#1073] [Q&A]
Operating system: e.g. Raspbian bullseye
Python version: e.g. 3.9.2
Pi model: e.g. Pi 3 Model B
GPIO Zero version: e.g. 1.6.2
Pin factory used: RPiGPIO
See http://rpf.io/gpzissue for information on how to find out these details

Please give us as much information about your issue as possible. Write your
code inside code blocks like so:

I'm using this code and print out an error:

```
from gpiozero import DistanceSensor
import time

while True:
    distance = DistanceSensor(echo=27, trigger=17)
    print(f"distance = {distance}")
```

error:

```
raise GPIOPinInUse('pin %s is already in use by %r' %
gpiozero.exc.GPIOPinInUse: pin 27 is already in use by <gpiozero.DistanceSensor object on pin GPIO27, pull_up=False>
```

I have no enough experience to fix. I'm reading issues but I can't understand where is the solution.
Thanks in advance

**Comment by bennuttall** [comment]
You're creating a new `DistanceSensor` object every time in the loop, so the second iteration will try to set up the same pins again. All you need to do is move the `distance = DistanceSensor()` line to before the loop:

```python
from gpiozero import DistanceSensor
import time

distance = DistanceSensor(echo=27, trigger=17)

while True:
    print(f"distance = {distance}")
```

**Comment by carlestecno** [comment]
Thanks @bennuttall,
I changed code but the  problem remains:

```
/usr/lib/python3/dist-packages/gpiozero/input_devices.py:855: PWMSoftwareFallback: For more accurate readings, use the pigpio pin factory.See https://gpiozero.readthedocs.io/en/stable/api_input.html#distancesensor-hc-sr04 for more info
  warnings.warn(PWMSoftwareFallback(
distance = <gpiozero.DistanceSensor object on pin GPIO27, pull_up=False>
```

**Comment by bennuttall** [comment]
That's a different message - and it's a warning not an error. You just need to read it :)

> PWMSoftwareFallback: For more accurate readings, use the pigpio pin factory. See https://gpiozero.readthedocs.io/en/stable/api_input.html#distancesensor-hc-sr04 for more info

**Comment by carlestecno** [comment]
Thanks, I will try to solve. If I get I will explain

**Comment by carlestecno** [comment]
Hi @bennuttall, missing the method of the class. 
Thanks again.
